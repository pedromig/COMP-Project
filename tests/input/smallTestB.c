int func(int);
int func(char);


int funcA(int a);
int funcA(void);
int funcA(void);

int funcB(void,int a);