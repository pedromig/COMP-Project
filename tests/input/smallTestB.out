Line 2, col 5: Conflicting types (got int(char), expected int(int))
Line 6, col 5: Conflicting types (got int(void), expected int(int))
Line 7, col 5: Conflicting types (got int(void), expected int(int))
Line 9, col 11: Invalid use of void type in declaration
===== Global Symbol Table =====
putchar	int(int)
getchar	int(void)
func	int(int)
funcA	int(int)

Program
..FuncDeclaration
....Int
....Id(func)
....ParamList
......ParamDeclaration
........Int
..FuncDeclaration
....Int
....Id(func)
....ParamList
......ParamDeclaration
........Char
..FuncDeclaration
....Int
....Id(funcA)
....ParamList
......ParamDeclaration
........Int
........Id(a)
..FuncDeclaration
....Int
....Id(funcA)
....ParamList
......ParamDeclaration
........Void
..FuncDeclaration
....Int
....Id(funcA)
....ParamList
......ParamDeclaration
........Void
..FuncDeclaration
....Int
....Id(funcB)
....ParamList
......ParamDeclaration
........Void
......ParamDeclaration
........Int
........Id(a)
